pkg/runtime/wfengine/activity.go

```diff
@@ -189,8 +201,16 @@ loop:
 				<-t.C
 			}
 			if completed {
+				elapsed := diag.ElapsedSince(start)
+				// Activity completed, record metrics
+				diag.DefaultWorkflowMonitoring.ExecutionCount(ctx, diag.Activity, diag.StatusSuccess)
+
+				// record the time taken to complete Activity execution
+				diag.DefaultWorkflowMonitoring.ExecutionLatency(ctx, diag.Activity, diag.StatusSuccess, elapsed)
```

Comment:
You can combine both into a common Record function, similar to this: https://github.com/dapr/dapr/blob/5050df5bf10835e9890b6b6519f5fa22e4736b28/pkg/diagnostics/component_monitoring.go#L210.